mlflow/pytorch/__init__.py

```diff
@@ -285,6 +287,108 @@ def gen_data():
     )
 
 
+@experimental
+def log_state_dict(state_dict, artifact_path, pickle_module=None, **kwargs):
+    """
+    Log a PyTorch model as an MLflow artifact for the current run.
+
+    :param state_dict: PyTorch model state dict to be saved.
+
+    :param artifact_path: Run-relative artifact path.
+
+    :param pickle_module: The module that PyTorch should use to serialize ("pickle") the specified
+                          ``pytorch_model``. This is passed as the ``pickle_module`` parameter
+                          to ``torch.save()``. By default, this module is also used to
+                          deserialize ("unpickle") the PyTorch model at load time.
+
+    :param kwargs: kwargs to pass to ``torch.save`` method.
+
+    .. code-block:: python
+        :caption: Example
+
+        import torch
+        import mlflow
+        import mlflow.pytorch
+        # X data
+        x_data = torch.Tensor([[1.0], [2.0], [3.0]])
+        # Y data with its expected value: labels
+        y_data = torch.Tensor([[2.0], [4.0], [6.0]])
+        # Partial Model example modified from Sung Kim
+        # https://github.com/hunkim/PyTorchZeroToAll
+        class Model(torch.nn.Module):
+            def __init__(self):
+               super().__init__()
+               self.linear = torch.nn.Linear(1, 1)  # One in and one out
+            def forward(self, x):
+                y_pred = self.linear(x)
+            return y_pred
+        # our model
+        model = Model()
+        criterion = torch.nn.MSELoss(size_average=False)
+        optimizer = torch.optim.SGD(model.parameters(), lr=0.01)
+        # Training loop
+        for epoch in range(500):
+            # Forward pass: Compute predicted y by passing x to the model
+            y_pred = model(x_data)
+            # Compute and print loss
+            loss = criterion(y_pred, y_data)
+            print(epoch, loss.data.item())
+            #Zero gradients, perform a backward pass, and update the weights.
+            optimizer.zero_grad()
+            loss.backward()
+            optimizer.step()
+
+        # After training
+        for hv in [4.0, 5.0, 6.0]:
+            hour_var = torch.Tensor([[hv]])
+            y_pred = model(hour_var)
+            print("predict (after training)",  hv, model(hour_var).data[0][0])
+        # log the model
+        with mlflow.start_run() as run:
+            mlflow.log_param("epochs", 500)
+            mlflow.pytorch.log_state_dict(model, "models")
+
+            # logging scripted module
+            scripted_pytorch_model = torch.jit.script(model)
+            mlflow.pytorch.log_state_dict(scripted_pytorch_model, "models")
+    """
+    pickle_module = pickle_module or mlflow_pytorch_pickle_module
+
+    with TempDir() as tmp:
+        local_path = tmp.path("model")
+        save_state_dict(
+            state_dict=state_dict, path=local_path, pickle_module=pickle_module, **kwargs
+        )
```

Comment:
I'm wondering if we need to `pickle_module` when logging a state dict. I think the reason that `torch.save` provides this argment is to allow users to use clouldpickle or dill which mitigates issues that pickle has when saving the entire model object. I'm investigating if the same issue still persists when saving a state_dict.

Reply:
cc @smurching

Reply:
I have removed the `pickle_module` argument. if needed we can add it as part of follow-up PR.